File: src/main/java/app/foxochat/dto/api/request/UserLoginDTO.java
package app.foxochat.dto.api.request;

import app.foxochat.constant.ValidationConstant;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Schema(name = "UserLogin")
public class UserLoginDTO {

    @NotNull(message = "Email/Username" + ValidationConstant.Messages.MUST_NOT_BE_NULL)
    @Size(min = ValidationConstant.Lengths.MIN, max = ValidationConstant.Lengths.EMAIL,
            message = "Email/Username" + ValidationConstant.Messages.WRONG_LENGTH)
    private String identity;

    @NotNull(message = "Password" + ValidationConstant.Messages.MUST_NOT_BE_NULL)
    @Size(min = ValidationConstant.Lengths.MIN, max = ValidationConstant.Lengths.PASSWORD,
            message = "Password" + ValidationConstant.Messages.WRONG_LENGTH)
    private String password;
}

File: src/main/java/su/foxogram/dto/api/request/UserLoginDTO.kt
package su.foxogram.dto.api.request

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import lombok.Getter
import lombok.Setter
import su.foxogram.constant.ValidationConstant

@Setter
@Getter
@Schema(name = "UserLogin")
data class UserLoginDTO(
    val email: @NotNull(message = "Email" + ValidationConstant.Messages.MUST_NOT_BE_NULL) @Size(
        min = ValidationConstant.Lengths.MIN,
        max = ValidationConstant.Lengths.EMAIL,
        message = ValidationConstant.Messages.EMAIL_WRONG_LENGTH
    ) @Pattern(
        regexp = ValidationConstant.Regex.EMAIL_REGEX,
        message = ValidationConstant.Messages.EMAIL_INCORRECT
    ) String? = null,

    val password: @NotNull(message = "Password" + ValidationConstant.Messages.MUST_NOT_BE_NULL) @Size(
        min = ValidationConstant.Lengths.MIN,
        max = ValidationConstant.Lengths.PASSWORD,
        message = ValidationConstant.Messages.PASSWORD_WRONG_LENGTH
    ) String? = null
)
